src/MapViewController.swift

```diff
@@ -76,7 +76,8 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
   var currentRouteMarker: TGMapMarkerId?
   open var shouldFollowCurrentLocation = false
   open var findMeButton = UIButton(type: .custom)
-  open var currentAnnotations: [PeliasMapkitAnnotation : TGMapMarkerId] = Dictionary()
+  open var annotationToMarkerId: [PeliasMapkitAnnotation : TGMapMarkerId] = Dictionary()
```

Comment:
Yeah the relationship is 1-1 so that could work.

FWIW `flatMap` is not THAT terrible for lookup performance. You can see the implementation of `flatMap` here - https://github.com/apple/swift/blob/master/stdlib/public/core/SequenceAlgorithms.swift.gyb#L673

I find it ironic that flatMap is literally just a for-in under the covers.